services/queue/src/main/java/tech/pegasys/pantheon/services/queue/RocksDbTaskQueue.java

```diff
@@ -37,13 +36,13 @@
   private final Options options;
   private final RocksDB db;
 
-  private final AtomicLong lastEnqueuedKey = new AtomicLong(0);
-  private final AtomicLong lastDequeuedKey = new AtomicLong(0);
+  private long lastEnqueuedKey = 0;
+  private long lastDequeuedKey = 0;
   private RocksIterator dequeueIterator;
-  private final AtomicLong oldestKey = new AtomicLong(0);
+  private long oldestKey = 0;
   private final Set<RocksDbTask<T>> outstandingTasks = new HashSet<>();
 
-  private final AtomicBoolean closed = new AtomicBoolean(false);
+  private boolean closed = false;
```

Comment:
Do these vars need to be `volatile`?

Reply:
No because every access to them is within a synchronized section.

Reply:
Still getting up to speed on threading obviously :) . I didn't realize `synchronized` causes all the vars to be updated from main memory.

Reply:
Yep, it has a read-fence on entry to the `synchronized` block and a write one on exit.  So as long as both the writing thread and the reading thread used synchronized there's no need for volatile.